src/Microsoft.Azure.WebJobs.Host.Storage/DynamicHostIdProvider.cs

```diff
@@ -15,68 +14,72 @@
 
 namespace Microsoft.Azure.WebJobs.Host.Executors
 {
-    // Negotiates host ID based on the assembly name of the first indexed method (persists a GUID for this purpose).
+    /// <summary>
+    /// Negotiates host ID based on the assembly name of the first job method (persists a GUID for this purpose).
+    /// </summary>
+    /// <remarks>
+    /// Because this SDK is used in both Console apps as well as WebApps, we can't rely on something simple like
+    /// Assembly.GetEntryAssembly() which won't work for WebApps. So we scan for the first assemlby with job methods
+    /// and use that.
+    /// </remarks>
     internal class DynamicHostIdProvider : IHostIdProvider
     {
-        private readonly LegacyConfig _storageAccountProvider;
-        private readonly IFunctionIndexProvider _getFunctionIndexProvider;
+        private readonly StorageAccountOptions _storageAccountOptions;
+        private readonly ITypeLocator _typeLocator;
 
-        public DynamicHostIdProvider(IOptions<LegacyConfig> storageAccountProvider, IFunctionIndexProvider getFunctionIndexProvider)
```

Comment:
The changes here were required to remove a circular DI dependency IHostIdProvider -> DynamicHostIdProvider -> FunctionIndexProvider -> SingletonManager -> IHostIdProvider.

We don't need the full index - we just need to look for the first job method, so I can simplify by using ITypeLocator